components/nimbus/src/stateful/nimbus_client.rs

```diff
@@ -704,6 +719,23 @@ impl NimbusClient {
         }
         Ok(())
     }
+
+    #[cfg(test)]
+    pub fn get_metrics_handler(&self) -> &TestMetrics {
+        let metrics = &*self.metrics_handler;
+        unsafe { std::mem::transmute::<&Box<dyn MetricsHandler>, &Box<TestMetrics>>(metrics) }
```

Comment:
I would get the &dyn MetricsHandler first and then transmute that into &TestMetrics, since the box is not important

Reply:
So I've done that here but Rust does not like that because they are sized differently 
<img width="1402" alt="image" src="https://github.com/mozilla/application-services/assets/4909307/6e36553d-430b-417d-be9e-425ee5b9329b">